### PR TITLE
Don't read contents of extra_files in memory at once

### LINKED ISSUE #329: Don't read contents of extra_files in memory at once
`extra_files` can specify a potentially large directory, which might not fit in memory all at once.
Freezeyt should only read file content in the task that saves that file.

### PR DESCRIPTION
**Pull request by encukou**
Fixes #329

### CODE REVIEW

File: tests/test_extra_files.py [hunk 1]
```diff
@@ -10,38 +12,26 @@ def test_simple():
...
-            'str': 'ábč',
-            'bytes': b'def',
-            'base64': {'base64': 'Z2hp'},
-            'copied': {
+            'str.txt': 'ábč',
+            'bytes.dat': b'def',
+            'base64.dat': {'base64': 'Z2hp'},
+            'copied.png': {
                 'copy_from': FIXTURES_PATH / 'app_with_extra_files/smile2.png'
             },
```
Comment (nappex) [inline tests/test_extra_files.py:25]: Take as example to explain the differ between url_part in application and path to the file, which should be used as static file.

File: freezeyt/freezer.py [hunk 2]
```diff
@@ -525,6 +550,14 @@ async def handle_content_task(self, task, content):
         self.done_tasks[task.path] = task
         self.call_hook('page_frozen', hooks.TaskInfo(task))
 
+    @needs_semaphore
+    async def handle_file_task(self, task, path):
```
Comment (nappex) [inline freezeyt/freezer.py:554]: Dont we consolidate this func and `handle_content_task`, to just handle task? Or it is another PR?

File: freezeyt/freezer.py [hunk 3]
```diff
@@ -384,6 +395,20 @@ def add_static_task(
             task.asyncio_task = asyncio_create_task(coroutine, name=task.path)
         return task
 
+    def add_file_task(
```
Comment (nappex) [inline freezeyt/freezer.py:398]: Dont we consolidate this func and `add_static_task` to just `add_task` or it was another PR?

**Review by nappex — APPROVED** [review]

File: freezeyt/freezer.py (hunk 3 above)
Comment (encukou) [reply freezeyt/freezer.py:398]: In another PR, when middleware serves the static files.

**Pull request merged**